ES6 class defined but never used

I'm testing the new ES6 support. The following code only passes if a property is assigned to the class in the constructor. Comment out that line and you get an error "Shoe is defined but never used" no-unused-vars

````
class Shoe {
    constructor() {
       Shoe.x = true;
    }
}
let s = new Shoe();
console.log(s);
````

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/9310837-es6-class-defined-but-never-used?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github).
</bountysource-plugin>

The code is:
````
class Shoe {
  constructor() {
    //Shoe.x = true;
  }
}
let s = new Shoe();
console.log(s);
````

The console output is:
````
eslint-demo.js
  1:6  error  Shoe is defined but never used  no-unused-vars

✖ 1 problem (1 error, 0 warnings)
````

--- a/lib/rules/no-unused-vars.js
+++ b/lib/rules/no-unused-vars.js
@@ -138,6 +138,7 @@ module.exports = function(context) {
                 var unresolvedRefs = globalScope.through.filter(isReadRef).map(function(ref) {
                     return ref.identifier.name;
                 });
+
                 for (i = 0, l = globalScope.variables.length; i < l; ++i) {
                     if (unresolvedRefs.indexOf(globalScope.variables[i].name) < 0 &&
                             !globalScope.variables[i].eslintJSXUsed && !isExported(globalScope.variables[i])) {
@@ -150,6 +151,12 @@ module.exports = function(context) {
                 if (unused[i].eslintExplicitGlobal) {
                     context.report(programNode, MESSAGE, unused[i]);
                 } else if (unused[i].defs.length > 0) {
+
+                    // TODO: Remove when https://github.com/estools/escope/issues/49 is resolved
+                    if (unused[i].defs[0].type === "ClassName") {
+                        continue;
+                    }
+
                     context.report(unused[i].identifiers[0], MESSAGE, unused[i]);
                 }
             }

--- a/tests/lib/rules/no-unused-vars.js
+++ b/tests/lib/rules/no-unused-vars.js
@@ -61,7 +61,9 @@ eslintTester.addRuleTest("lib/rules/no-unused-vars", {
         { code: "var who = \"Paul\";\nmodule.exports = `Hello ${who}!`;", ecmaFeatures: { templateStrings: true }},
         { code: "export var foo = 123;", ecmaFeatures: { modules: true }},
         { code: "export function foo () {}", ecmaFeatures: { modules: true }},
-        { code: "export class foo {}", ecmaFeatures: { modules: true, classes: true }}
+        { code: "export class foo {}", ecmaFeatures: { modules: true, classes: true }},
+        { code: "class Foo{}; var x = new Foo(); x.foo()", ecmaFeatures: { classes: true }},
+        "function Foo(){}; var x = new Foo(); x.foo()"
     ],
     invalid: [
         { code: "var a=10", errors: [{ message: "a is defined but never used", type: "Identifier"}] },

